include/mlc/base/traits_dtype.h

```diff
@@ -63,6 +63,7 @@ template <> struct TypeTraits<DLDataType> {
 };
 
 inline DLDataType DataTypeFromStr(const char *source) { return ::mlc::Lib::DataTypeFromStr(source); }
+inline std::string DataTypeToStr(DLDataType dtype) { return TypeTraits<DLDataType>::__str__(dtype); }
```

Comment:
Can we extend to new data types such as fp8 and fp4 easily? dlpack has been outdated for a long time, and fp8 is common practice in training/inference of foundation models.

Reply:
Yeah it’s already supported: https://github.com/mlc-ai/mlc-python/blob/main/tests/python/test_core_dtype.py#L115